engine/runtime/src/main/java/org/enso/interpreter/runtime/error/DataflowError.java

```diff
@@ -53,10 +53,11 @@ public final class DataflowError extends AbstractTruffleException {
    * @return a new dataflow error
    */
   public static DataflowError withDefaultTrace(
-      State state, Object payload, Node location, HasContextEnabledNode hasContextEnabledNode) {
+      Object payload, Node location, HasContextEnabledNode hasContextEnabledNode) {
     assert payload != null;
     var ensoCtx = EnsoContext.get(location);
     var dataflowStacktraceCtx = ensoCtx.getBuiltins().context().getDataflowStackTrace();
+    State state = ensoCtx.currentState();
     boolean attachFullStackTrace =
         state == null
```

Comment:
Was this condition correct to begin with? When there is no `state`, then attachFullStackTrace!? The default should be to don't attach full stacktraces to `DataflowError`s as that's slow.

We have a [RuntimeErrorsTest](https://github.com/enso-org/enso/actions/runs/13161158904/job/36729996193?pr=12233#step:8:3882) which can easily be fixed by setting `state = null`.

However I think it is wrong. There should be no _full stack trace_ attached in the test (unless the test asks for full stack trace somehow; but it doesn't). 

https://github.com/enso-org/enso/blame/86520e3a9e97944f893ad51fd1810e8be1f80ad1/engine/runtime/src/main/java/org/enso/interpreter/runtime/error/DataflowError.java#L62

@4e6, you seem to be the author of the condition. why it is the way it is?